monocle/webapp.py

```diff
@@ -152,8 +178,17 @@ def query(name):
 
 
 @cache.memoize(timeout=CACHE_TIMEOUT)
-def do_query(index, repository_fullname, args, name):
+def do_query(index, repository_fullname, args, name, projects=None):
     params = utils.set_params(args)
+
+    if params["project_definition"] and not projects:
+        return (
+            "There are no project definion set in "
+            "config file: %s" % request.get("index"),
+            404,
+        )
+
+    params["config_project_definitions"] = projects
```

Comment:
Not sure why we need that. We already have the endpoint that is api/0/projects so that's enough.